### PR TITLE
fixed the software resized texture warp problem.

### PR DESCRIPTION
**Pull request by madame-rachelle**
This is incomplete, texture mipmaps need to be generated and work with this.

https://forum.zdoom.org/viewtopic.php?f=56&t=63919

### CODE REVIEW

File: src/rendering/swrenderer/textures/warptexture.cpp [hunk 1]
```diff
@@ -60,8 +61,9 @@ const uint32_t *FWarpTexture::GetPixelsBgra()
 	if (time != GenTime[2])
 	{
 		auto otherpix = FSoftwareTexture::GetPixelsBgra();
-		WarpedPixelsRgba.Resize(GetWidth() * GetHeight());
-		WarpBuffer(WarpedPixelsRgba.Data(), otherpix, GetWidth(), GetHeight(), WidthOffsetMultiplier, HeightOffsetMultiplier, time, mTexture->shaderspeed, bWarped);
+		WarpedPixelsRgba.Resize(GetWidth() * GetHeight() * gl_texture_hqresizemult * gl_texture_hqresizemult * 4 / 3 + 1);
```
Comment (alexey-lysiuk) [inline src/rendering/swrenderer/textures/warptexture.cpp]: That `+ 1` looks a bit suspicious. Doesn’t lower mop level require some alignment, like 4 or 16 bytes?

**Comment by drfrag666** [comment]
What about just leaving the *2 to be safe? The parameter is an unsigned int and by default gl_texture_hqresizemult is 1, 4/3 is also 1 so you're just adding 1. I've looked at the Resize implementation and it's a bit complicated, it sure adds "some" elements. :)
 That unless i missed something.

File: src/rendering/swrenderer/textures/warptexture.cpp (hunk 1 above)
Comment (madame-rachelle) [reply src/rendering/swrenderer/textures/warptexture.cpp]: I put it in there in order to ensure that rounding errors would not lower the amount of space required for mipmapped textures. After some calculation I realized that for every level of mipmap, that the required amount for mipmaps will *never* exceed 1/3rd of the total texture size, even if mipmaps are continually generated off of the previous one, but that wholly depends on how the mipmaps were stored in the TArray.

As for the alignment - I was not too sure about that, myself. However, I do notice that this allocation can be completely arbitrary, i.e. if a texture is 3x13, it supposedly should work. If I am wrong about this, feel free to correct me, but I suspect there's some intermediary code occurring with the resize that makes this safe.

Comment (alexey-lysiuk) [reply src/rendering/swrenderer/textures/warptexture.cpp]: Actually, I have no idea how mipmaps work in software renderer. If they don't require any alignment then `+ 1` is probably OK.

**Comment by madame-rachelle** [comment]
I've gone ahead and put this in. It is a bug fix and I am fairly confident in the code, and this is after all a bug fix - if there are issues, this can be reversed (it's squashed), or the issues can be fixed.

**Pull request merged**